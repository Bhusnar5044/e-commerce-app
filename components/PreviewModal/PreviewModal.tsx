'use client';

import { memo } from 'react';

import Gallery from '@/components/Gallery';
import Info from '@/components/Info';
import Modal from '@/components/ui/Modal';
import { onClose } from '@/redux/features/modalSlice';
import { useAppDispatch, useAppSelector } from '@/redux/store';

export const PreviewModal = memo(() => {
    const { isOpen, data } = useAppSelector((state) => state.modal);
    const dispatch = useAppDispatch();

    if (!data) {
        return null;
    }

    const handleOnClose = () => dispatch(onClose());

    return (
        <Modal open={isOpen} onClose={handleOnClose}>
            <div className="grid w-full grid-cols-1 items-start gap-x-6 gap-y-8 sm:grid-cols-12 lg:gap-x-8">
                <div className="sm:col-span-4 lg:col-span-5">
                    <Gallery images={data.images} />
                </div>
                <div className="sm:col-span-8 lg:col-span-7">
                    <Info data={data} />
                </div>
            </div>
        </Modal>
    );
});

PreviewModal.displayName = 'PreviewModal';
