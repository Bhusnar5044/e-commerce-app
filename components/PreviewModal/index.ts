export { PreviewModal as default } from './PreviewModal';
