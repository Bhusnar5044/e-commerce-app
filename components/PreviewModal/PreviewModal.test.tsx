import { renderWithStore } from '@test-utils';
import { PreviewModal } from './PreviewModal';

describe('PreviewModal', () => {
    it('should render properly', () => {
        const { container } = renderWithStore(<PreviewModal />);
        expect(container).toMatchSnapshot();
    });
});
