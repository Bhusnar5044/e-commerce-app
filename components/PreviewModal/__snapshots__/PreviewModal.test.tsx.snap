// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewModal should render properly 1`] = `<div />`;
